pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -634,6 +638,59 @@ func init() {
 	}
 
 	err = InfraSAs.addServiceAccount(
+		InfraPetSetControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: PetSetControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// PetSetController.podCache.ListWatch
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("list", "watch"),
+					Resources: sets.NewString("pods"),
+				},
+				// PetSetController.cache.ListWatch
+				{
+					APIGroups: []string{apps.GroupName},
+					Verbs:     sets.NewString("list", "watch"),
+					Resources: sets.NewString("petsets"),
+				},
+				// PetSetController.petClient
```

Comment:
I'm not convinced that we should split out rules like this.  We never come backto them and they're hard to read.  How about one rule per resource.